File: spec/javascript/test_type_int_unary_op.js
var assert = require('assert');
var testHelper = require('testHelper');

testHelper('TypeIntUnaryOp', 'src/fixed_struct.bin', function(r) {
  assert.equal(r.valueS2, 0x4150);
  assert.equal(r.valueS8, 0x4150ffff312d4b43);
  assert.equal(r.unaryS2, -0x4150);
  assert.equal(r.unaryS8, -0x4150ffff312d4b43);
});

File: spec/java/src/io/kaitai/struct/spec/TestTypeIntUnaryOp.java
package io.kaitai.struct.spec;

import io.kaitai.struct.testformats.TypeIntUnaryOp;
import org.testng.annotations.Test;

import static org.testng.Assert.assertEquals;
import static org.testng.Assert.assertTrue;

public class TestTypeIntUnaryOp extends CommonSpec {
    @Test
    public void testTypeIntUnaryOp() throws Exception {
        TypeIntUnaryOp r = TypeIntUnaryOp.fromFile(SRC_DIR + "fixed_struct.bin");

        assertEquals(r.valueS2(), (short) 0x4150);
        assertEquals(r.valueS8(), 0x4150ffff312d4b43L);
        assertTrue(r.unaryS2() instanceof Long);
        assertTrue(r.unaryS8() instanceof Long);
        assertEquals(r.unaryS2().shortValue(), (short) -0x4150);
        assertEquals(r.unaryS8().longValue(), -0x4150ffff312d4b43L);
    }
}
